Feature/swift3

sEmoto0808 opened the pull request:
### 概要
Swift2対応版からSwift3対応版へ修正

### 対応リスト

### 更新内容
#### 新規追加
①不要なコードの削除
　不要なコードを削除する旨を記載
②ジェネリクス
　可能な限りジェネリクスを活用する旨を記載

#### 変更
①コーディング規約のバージョン
②データ型の変更
　・AnyObject → Any に修正
　・NSURL→URL
　・NSDate→Date
　・CGColor→cgColor（記述をSwift3用に変更）
③アクセス制御に「open」と「fileprivate」を追加
④ファイルフォーマットのファイル内順序を修正
⑤弱参照の項目にweakとunownedの違いを記載し、例を修正
⑥ クロージャーの規則と例を修正
⑦型推論の例を追加
⑧継承よりもプロトコルを活用する旨の記載を追加

### 特記事項
なし

Commit:
[fix] バージョン修正

Commit:
[modify] Swift3に対応するために以下を修正
・コーディング規約のバージョン
・アクセス制御にfileprivateを追加
・AnyObject → Any に修正
・ファイルフォーマットのファイル内順序を修正
・NSURL→URL、NSDate→Date、CGColor→cgColor

Commit:
[modify] 以下の追加と変更
＜追加項目＞
不要なコード削除の説明と例
＜変更項目＞
①弱参照の理由と例
②クロージャーの説明と例

Commit:
[fix] markdownで改行

Commit:
[add] 型推論の例を追加

Commit:
[add] 可能な限りプロトコルとジェネリクスを利用するよう記載を追加

stv-ekushida commented on README.md:
````diff
@@ -939,12 +897,18 @@ if let bindedSome = some {
 
 ```swift
 someFunc { [weak self] in
+  guard let self = self {
````
バックコートなくても、コンパイル通るんでしたっけ？

```
guard let `self` = self {
    return
}
```

sEmoto0808 replied:
テキストに写す時に漏れていました。
ありがとうございます。

stv-ekushida commented on README.md:
```diff
@@ -939,12 +897,18 @@ if let bindedSome = some {
+
 **例**
 
 良い例
```
悪い例も書いて下さい、

sEmoto0808 replied:
悪い例追加しました。

stv-ekushida commented on README.md:
```diff
@@ -939,12 +897,18 @@ if let bindedSome = some {
 
 nilアクセスの可能性をできるだけ下げるため。
 
+[weak self]・・・弱参照先がメモリ解放されている場合にnilになる（Optional型と同様にアンラップが必要）  
```
メモリ管理という段落をつくったら如何でしょうか？

sEmoto0808 replied:
メモリ管理の段落を追加して、そちらに記述を移しました。

stv-ekushida commented on README.md:
```diff
@@ -981,8 +945,11 @@ let someInt = try! someErrorThrowMethod()
+③後置クロージャを持つメソッドでメソッドチェーンで記述する場合は、スペースを入れる or 改行するで統一する。  
 
 **理由**
 
```
理由も書いて下さい。

sEmoto0808 replied:
理由追加しました。

stv-ekushida replied:
👍

Commit:
[fix] 指摘事項の反映
・guard let self = self { return }　→　guard let `self` = self { return }
・メモリ管理の段落追加
・メモリ管理のの項目に悪い例を追加
・クロージャーの項目に理由を追加

Commit:
[fix] レイアウト修正

Commit:
[add] 型推論の例を追加

Commit:
[fix] レイアウト修正

stv-ekushida commented on README.md:
````diff
@@ -944,109 +921,194 @@ nilアクセスの可能性をできるだけ下げるため。
+
+```swift
+someFunc { [weak self] in
+  guard let `self` = self {
````
バッククオートでselfに代入できるのはバグだという情報もありますから、
下記のようにしてください。

```
guard let weakSelf = self {
    return
}
let sampleTuple = weakSelf.makeTuple(a: "A", b: "B")

```

y-okudera replied:
バグかどうかは分かりませんが、
バッククォートはSwiftの予約語と同名を付けるときに使用するものです。
予約語と同名は避けるべきです。

参考

```
If you need to give a constant or variable the same name as a reserved Swift keyword, surround the keyword with backticks (`) when using it as a name. However, avoid using keywords as names unless you have absolutely no choice.
```

https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/TheBasics.html#//apple_ref/doc/uid/TP40014097-CH5-ID313

sEmoto0808 replied:
修正しました。

stv-ekushida commented on README.md:
```diff
@@ -944,109 +921,194 @@ nilアクセスの可能性をできるだけ下げるため。
+}
+
+UIView.animate(withDuration: 10.0) {
+        print("animate")
```
細かいですが、インデントがおかしいです。
全体的にインデントを見直してください。

sEmoto0808 replied:
修正しました。

stv-ekushida commented on README.md:
```diff
@@ -944,109 +921,194 @@ nilアクセスの可能性をできるだけ下げるため。
+}
+
+UIView.animate(withDuration: 10.0, animations:{
+        print("animate")  
```
インデントがおかしいです。

stv-ekushida commented on README.md:
````diff
@@ -944,109 +921,194 @@ nilアクセスの可能性をできるだけ下げるため。
+
+```swift
+override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
````
インデントがおかしいです。

Commit:
[fix] 以下の内容を修正
・重複していた項目の削除（項目名：使われていないコードの削除）
・`self` → weakSelf に修正
・インデントの修正（PCで閲覧した場合）

Commit:
[fix] インデントの修正
